integration/service/src/main/java/com/cerner/beadledom/integration/service/resource/HelloWorldResource.java

```diff
@@ -0,0 +1,43 @@
+package com.cerner.beadledom.integration.service.resource;
+
+import com.cerner.beadledom.integration.api.model.HelloWorldDto;
+import com.cerner.beadledom.jaxrs.GenericResponse;
+import com.cerner.beadledom.jaxrs.GenericResponses;
+import com.cerner.beadledom.pagination.OffsetPaginatedList;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import java.util.ArrayList;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Server implementation of an offset based paginated list endpoint.
+ *
+ * @author Nick Behrens
+ */
+@Api(value = "/hello", description = "Retrieve hello world data")
+@Path("/hello")
+public class HelloWorldResource {
```

Comment:
I think we can just extend the resource class from the api project?

Reply:
With the current way they are set up, we cannot. The api project implementation of `getHelloWorld` returns an OffsetPaginatedListHelloWorldDto which implements the `OffsetPaginatedListDto` whereas the service resource returns a type of `OffsetPaginatedList`. That being said, I might be missing something obvious with how to set it up.

Reply:
This is how it works in red-tape also for v2 resources. Whats the reason for this discrepancy?

Reply:
It's because the `OffsetPaginatedList` contains logic to generate the next and previous links. The Dto however doesn't contain this logic because it doesn't need to. We return the `OffsetPaginatedList` because the links for the returned object matter but for the dto representation they don't.

Reply:
I'm not entirely sure, the docs for `OffsetPagiantedListDto` say that "The `OffsetPaginatedListDto` is used on the client side instead of {@link OffsetPaginatedList} to circumvent type erasure."